stats/csv/collector_test.go

```diff
@@ -336,6 +345,15 @@ func TestNew(t *testing.T) {
 				"tag2",
 			},
 		},
+		{
+			fname: "name.csv.gz",
+			resTags: []string{
+				"tag1", "tag3",
+			},
+			ignoredTags: []string{
+				"tag2",
+			},
+		},
```

Comment:
Could you add a more functional test than this configuration one? Something like `TestRunCollect` that runs the collector and checks the output file. Maybe you can parametrize that one and make `compressed` and `uncompressed` subtests.

Reply:
Sure, will give this a try

Reply:
@imiric I have added these tests and the `flush` changes.